core/src/main/java/com/linecorp/armeria/common/logging/DefaultRequestLog.java

```diff
@@ -845,8 +844,9 @@ private void endRequest0(@Nullable Throwable requestCause) {
 
     private void endRequest0(@Nullable Throwable requestCause, long requestEndTimeNanos) {
         final int flags;
-        if (requestCause == null && requestContentDeferred) {
-            flags = RequestLogProperty.FLAGS_REQUEST_COMPLETE_WITHOUT_CONTENT;
+        final int deferredFlags = this.deferredFlags;
+        if (requestCause == null && deferredFlags != 0) {
```

Comment:
`deferredFlags` doesn't need to be read when `requestCause` is not `null`.

Reply:
🙆‍♂